linera-views/src/random.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use rand::{Rng, SeedableRng};
```

Comment:
Should this be tagged as `with_testing`?
Can we avoid exposing all the public methods below?

Reply:
The problem arises because we cannot tag the member function with `with_testing` because of the `trait_variant` macro.

Reply:
So, I introduced the `TestKeyValueStore` trait. This allowed me to make many functions under the `with_testing` umbrella.